pkg/cmd/autocomplete.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package cmd
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"github.com/spf13/cobra"
+)
+
+var CompletionCmd = &cobra.Command{
+	Use:     "completion [bash|zsh|fish|powershell]",
+	Short:   "Adds completion script for your shell enviornment",
+	Args:    cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		shell := args[0]
+		homeDir, err := os.UserHomeDir()
+		if err != nil {
+			fmt.Printf("Error finding user home directory: %s\n", err)
+			return
+		}
+
+		var filePath, profilePath string
+		switch shell {
+		case "bash":
+			filePath = filepath.Join(homeDir, ".daytona.completion_script.bash")
+			profilePath = filepath.Join(homeDir, ".bashrc")
+		case "zsh":
+			filePath = filepath.Join(homeDir, ".daytona.completion_script.zsh")
+			profilePath = filepath.Join(homeDir, ".zshrc")
+		case "fish":
+			filePath = filepath.Join(homeDir, ".config", "fish", "daytona.completion_script.fish")
+			profilePath = filepath.Join(homeDir, ".config", "fish", "config.fish")
+		case "powershell":
+			filePath = filepath.Join(homeDir, "daytona.completion_script.ps1")
+			profilePath = filepath.Join(homeDir, "Documents", "WindowsPowerShell", "Microsoft.PowerShell_profile.ps1")
+		default:
+			fmt.Println("Unsupported shell type. Please use bash, zsh, fish, or powershell.")
+			return
+		}
+
+		file, err := os.Create(filePath)
+		if err != nil {
+			fmt.Printf("Error creating completion script file: %s\n", err)
+			return
+		}
+		defer file.Close()
+
+		switch shell {
+		case "bash":
+			err = cmd.Root().GenBashCompletion(file)
+		case "zsh":
+			err = cmd.Root().GenZshCompletion(file)
+		case "fish":
+			err = cmd.Root().GenFishCompletion(file, true)
+		case "powershell":
+			err = cmd.Root().GenPowerShellCompletionWithDesc(file)
+		}
+
+		if err != nil {
+			fmt.Printf("Error generating completion script: %s\n", err)
+			return
+		}
+
+		sourceCommand := fmt.Sprintf("source %s\n", filePath)
+		if shell == "powershell" {
+			sourceCommand = fmt.Sprintf(". %s\n", filePath)
+		}
+
+		alreadyPresent := false
+		profile, err := os.Open(profilePath)
+		if err == nil {
+			scanner := bufio.NewScanner(profile)
+			for scanner.Scan() {
```

Comment:
I suggest making this a bit more clean simply by using `os.ReadFile` and `strings.Contains`. You can see an example usage in our code [here](https://github.com/daytonaio/daytona/blob/main/cmd/daytona/config/ssh_file.go#L114)